### PR TITLE
Grw-1413 Chore inline blocks

### PR DESCRIPTION
**Pull request by KarnellSchultz**
## Describe your changes

Refactored storyblock components so they no longer use the generic `<StoryblokComponent />`.

## Justify why they are needed

This will make it a bit more difficult for components to be placed in unintended locations. For example if a `<HeaderBlock />` is only expected a `<h2 />` then it would no longer be possible to place another component inside of the `<HeaderBlock />`.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by KarnellSchultz** [comment]
There are a few places where it makes sense to use `<StoryblokComponent />` like Page blocks or blocks that can take a wide variety of blocks.

File: apps/store/src/blocks/HeroBlock.tsx [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 }>
 
 export const HeroBlock = ({ blok }: HeroBlockProps) => {
   return (
     <HeroSection {...storyblokEditable(blok)} bgImage={blok.background.filename}>
       <div>
-        {blok.content.map((nestedBlock) => (
-          <StoryblokComponent blok={nestedBlock} key={nestedBlock._uid} />
-        ))}
+        {blok.content.map((nestedBlock) => {
```
Comment (simonauner) [inline apps/store/src/blocks/HeroBlock.tsx:17]:
```suggestion
        {blok.content.map((nestedBlock) => <HeadingBlock blok={nestedBlock} key={nestedBlock._uid} />)}
```

**Review by simonauner — APPROVED** [review]
Nice refactoring 🙌

File: apps/store/src/blocks/HeroBlock.tsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
 }>
 
 export const HeroBlock = ({ blok }: HeroBlockProps) => {
   return (
     <HeroSection {...storyblokEditable(blok)} bgImage={blok.background.filename}>
       <div>
-        {blok.content.map((nestedBlock) => (
-          <StoryblokComponent blok={nestedBlock} key={nestedBlock._uid} />
-        ))}
+        {blok.content.map((nestedBlock) => <HeadingBlock blok={nestedBlock} key={nestedBlock._uid} />)}
```
Comment (alebedev) [inline apps/store/src/blocks/HeroBlock.tsx:17]: Let's also filter on client side as additional safeguard against incorrect Storyblok config

```
const contentBlocks = filterBlocksByType(blok.content, HeadingBlock.blockName)
```

**Comment by alebedev** [comment]
From failed Vercel deployment you can see that yarn.lock needs updating.  Since you did not change any deps, let's investigate if our main branch is healthy

File: apps/store/src/blocks/HeroBlock.tsx (hunk 2 above)
Comment (KarnellSchultz) [reply apps/store/src/blocks/HeroBlock.tsx:17]: totally forgot about that, good call 👍

File: apps/store/src/blocks/AccordionBlock.tsx [hunk 3]
```diff
@@ -2,12 +2,12 @@ import styled from '@emotion/styled'
...
 import { AccordionItemBlock, AccordionItemBlockProps } from '@/blocks/AccordionItemBlock'
 import * as Accordion from '@/components/Accordion/Accordion'
-import { SbBaseBlockProps } from '@/services/storyblok/storyblok'
+import { ExpectedBlockType, SbBaseBlockProps } from '@/services/storyblok/storyblok'
 import { filterByBlockType } from '@/services/storyblok/Storyblok.helpers'
 
 type Props = SbBaseBlockProps<{
   title?: string
-  items: Array<AccordionItemBlockProps['blok']>
+  items: ExpectedBlockType<AccordionItemBlockProps>
```
Comment (alebedev) [inline apps/store/src/blocks/AccordionBlock.tsx:10]: Nice!

File: apps/store/src/services/storyblok/storyblok.ts [hunk 4]
```diff
@@ -27,6 +27,11 @@ export type SbBaseBlockProps<T> = {
   blok: SbBlokData & T
 }
 
+export type ExpectedBlockType<T> = T extends { blok: SbBlokData }
```
Comment (simonauner) [inline apps/store/src/services/storyblok/storyblok.ts:30]: Cool! TS Master Level 1000!

Comment (guilhermespopolin) [inline apps/store/src/services/storyblok/storyblok.ts:30]: Have never seen that _conditional types_ feature. You never stop learn :)

**Review by guilhermespopolin — APPROVED** [review]
Well done :)

File: apps/store/src/services/storyblok/storyblok.ts (hunk 4 above)
Comment (KarnellSchultz) [reply apps/store/src/services/storyblok/storyblok.ts:30]: lolol - perhaps level 4 or 5. One day I'll get to level 1000 https://media.giphy.com/media/l0Exe5NhestV2TfC8/giphy.gif

Comment (KarnellSchultz) [reply apps/store/src/services/storyblok/storyblok.ts:30]: Yeah it's really one of the coolest things you can do.

File: apps/store/src/services/storyblok/Storyblok.helpers.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { SbBlokData } from '@storyblok/react'
 
 export const filterByBlockType = <BlockData extends SbBlokData>(
-  blocks: BlockData[],
+  blocks: BlockData[] = [],
```
Comment (KarnellSchultz) [inline apps/store/src/services/storyblok/Storyblok.helpers.ts:4]: Added So this does not throw an error when the argument is optional.

**Pull request merged**